extensions/interactions/MultipleChoiceInput/directives/oppia-interactive-multiple-choice-input.directive.ts

```diff
@@ -77,8 +77,31 @@ angular.module('oppia').directive('oppiaInteractiveMultipleChoiceInput', [
           };
 
           ctrl.$onInit = function() {
-            ctrl.choices = HtmlEscaperService.escapedJsonToObj(
+            var showChoicesInShuffledOrder = (
+              $attrs.showChoicesInShuffledOrderWithValue === 'true');
+            var choicesWithValue = HtmlEscaperService.escapedJsonToObj(
               $attrs.choicesWithValue);
+            var choicesWithIndex = choicesWithValue.map(
+              function(value, originalIndex) {
+                return {originalIndex: originalIndex, value: value};
+              }
+            );
+            var shuffleChoices = function(choices) {
+              var currentIndex = choices.length - 1;
+              var temporaryValue = null;
+              var randomIndex = null;
+
+              for (; currentIndex >= 0; currentIndex--) {
```

Comment:
You should probably initialize currentIndex here instead so that it's more clearly a local var.

You can also define temporaryValue and randomIndex inside the for loop, since they are not needed outside it.

In short, you shouldn't need lines 90-92.

Reply:
Done.